rpc/handlers.go

```diff
@@ -1169,6 +1181,31 @@ func (h *Handler) AddTransaction(ctx context.Context, tx BroadcastedTransaction)
 	}, nil
 }
 
+func (h *Handler) addToMempool(tx *BroadcastedTransaction) (*AddTxResponse, *jsonrpc.Error) {
+	userTxn, userClass, _, err := adaptBroadcastedTransaction(tx, h.bcReader.Network())
+	if err != nil {
+		return nil, ErrInternal.CloneWithData(err.Error())
+	}
+	err = h.memPool.Push(&mempool.BroadcastedTransaction{
+		Transaction:   userTxn,
+		DeclaredClass: userClass,
+	})
+	if err != nil {
+		return nil, ErrValidationFailure.CloneWithData(err.Error())
+	}
+
+	res := &AddTxResponse{TransactionHash: userTxn.Hash()}
+	if tx.Type == TxnDeployAccount {
+		res.ContractAddress = core.ContractAddress(&felt.Zero, tx.ClassHash, tx.ContractAddressSalt, *tx.ConstructorCallData)
+	} else if userClass != nil {
```

Comment:
Is this equivalent to `tx.Type == TxnDeclare || tx.Type == TxnDeploy`?

Reply:
> tx.Type == TxnDeclare

Probably only this, `TxnDeploy` is not supported anymore.